### PR TITLE
NDW-5

### PR DESCRIPTION
**Pull request by brettvniekerk**
- [Link to ticket](https://dltx.atlassian.net/jira/software/projects/NDW/boards/76?selectedIssue=NDW-5)
- Added sudo application

### CODE REVIEW

File: src/components/ContractPanel.tsx [hunk 1]
```diff
@@ -12,17 +12,26 @@ const ContractPanel: React.FC<Props> = (props) => {
...
 
+  useEffect(() => {
+    contract
+      .owner()
+      .then((owner) => {
+        if (owner === address) return setOwner(true);
+      })
+      .catch((err) => setError(err));
+  }, [address]);
+
```
Comment (Preston-Harrison) [inline src/components/ContractPanel.tsx:22]: fetch the contract owner once, on load. then just `const isOwner = address === owner`

File: src/components/ContractPanel.tsx [hunk 2]
```diff
@@ -12,17 +12,26 @@ const ContractPanel: React.FC<Props> = (props) => {
...
   const authenticate = async () => {
     try {
       setLoading(true);
 
-      const balance = await contract.balanceOf(address);
+      const balance = await contract.balanceOf(
+        "0xe33c15ddf8d2afad6e17f9935a1af90c21ca7b0d"
```
Comment (Preston-Harrison) [inline src/components/ContractPanel.tsx]: magic address? where is this coming from

**Review by OptimusOpus — DISMISSED** [review]
LGTM

File: src/components/MemberPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from "react";
+import { Button, SudoApplicationModal } from "../components";
+
+type Props = {
+  address: string;
+};
+
+const OwnerPanel: React.FC<Props> = (props) => {
```
Comment (Preston-Harrison) [inline src/components/MemberPanel.tsx]: this is called OwnerPanel, but is exported in the file MemberPanel

File: src/components/SudoApplicationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  isModalOpen: boolean;
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  address: string;
+};
+
+const SudoApplicationModal: React.FC<Props> = (props) => {
+  const { isModalOpen, setIsModalOpen, address } = props;
+
+  const sendApplication = async () => {
+    window.location.replace(
```
Comment (Preston-Harrison) [inline src/components/SudoApplicationModal.tsx:14]: Might make more sense for this link to be in an `<a>` tag

**Review by Preston-Harrison — CHANGES_REQUESTED** [review]

File: src/components/MemberPanel.tsx (hunk 3 above)
Comment (brettvniekerk) [reply src/components/MemberPanel.tsx]: oops, good catch

File: src/components/SudoApplicationModal.tsx (hunk 4 above)
Comment (brettvniekerk) [reply src/components/SudoApplicationModal.tsx:14]: Wanted to keep the button styling without rebuilding it for an <a> tag - makes no end user difference

File: src/components/ContractPanel.tsx (hunk 1 above)
Comment (brettvniekerk) [reply src/components/ContractPanel.tsx:22]: 👍

**Review by OptimusOpus — APPROVED** [review]

**Review by Preston-Harrison — APPROVED** [review]

**Pull request merged**